File: core/src/main/java/com/mcreater/amclcore/config/ConfigParser.java
package com.mcreater.amclcore.config;

import com.mcreater.amclcore.annotations.ConfigModel;
import com.mcreater.amclcore.util.IOStreamUtil;

import java.io.File;
import java.io.IOException;

import static com.mcreater.amclcore.util.JsonUtil.GSON_PARSER;

public class ConfigParser {
    private ConfigParser() {
    }

    public static <T> T parse(File configFile, Class<T> clazz) throws IOException {
        if (clazz.getAnnotation(ConfigModel.class) != null)
            return GSON_PARSER.fromJson(IOStreamUtil.toReader(configFile), clazz);
        else throw new IOException("Not a config model");
    }

    public static <T> void write(File configFile, T model) throws IOException {
        GSON_PARSER.toJson(model, IOStreamUtil.toWriter(configFile));
    }
}

File: core/src/main/java/com/mcreater/amclcore/config/ConfigParser.kt
package com.mcreater.amclcore.config

import com.mcreater.amclcore.annotations.ConfigModel
import com.mcreater.amclcore.util.IOStreamUtil
import com.mcreater.amclcore.util.JsonUtil
import java.io.File
import java.io.IOException


class ConfigParser {
    companion object {
        @JvmStatic
        @Throws(IOException::class)
        fun <T> parse(configFile: File?, clazz: Class<T>): T {
            return if (clazz.getAnnotation(ConfigModel::class.java) != null) JsonUtil.GSON_PARSER.fromJson(
                IOStreamUtil.toReader(
                    configFile
                ), clazz
            ) else throw IOException("Not a config model")
        }

        @JvmStatic
        @Throws(IOException::class)
        fun <T> write(configFile: File?, model: T) {
            JsonUtil.GSON_PARSER.toJson(model, IOStreamUtil.toWriter(configFile))
        }
    }
}

